bciers/apps/reporting/src/app/components/finalReview/reviewDataFactory/factory.ts

```diff
@@ -22,23 +27,48 @@ export type ReviewDataFactoryItem = (
 
 export default async function reviewDataFactory(
   versionId: number,
+  flow: ReportingFlowDescription,
 ): Promise<ReviewData[]> {
   // Fetch facilities for this report version
   const listFacilities = (await getOperationFacilitiesList(versionId)) || [];
   const currentFacilities = listFacilities.current_facilities;
   const facilityId = currentFacilities[0].facility_id;
-  const operationType = currentFacilities.length > 1 ? "LFO" : "SFO";
 
-  let reviewData: ReviewData[] = [
-    ...(await operationReviewFactoryItem(versionId, facilityId)),
-    ...(await personResponsibleFactoryItem(versionId, facilityId)),
-    ...(await facilityActivitiesFactoryItem(versionId, currentFacilities)),
-    ...(await additionalReportingDataFactoryItem(versionId, facilityId)),
-    ...(operationType === "LFO"
-      ? await operationEmissionSummaryFactoryItem(versionId, facilityId)
-      : []),
-    ...(await complianceSummaryFactoryItem(versionId, facilityId)),
-  ];
+  // Mapping from ReportingPage to the corresponding factory function
+  const factoryMapping: Partial<
+    Record<ReportingPage, (...args: any[]) => Promise<ReviewData[]>>
+  > = {
+    [ReportingPage.ReviewOperationInfo]: operationReviewFactoryItem,
+    [ReportingPage.PersonResponsible]: personResponsibleFactoryItem,
+    [ReportingPage.Activities]: facilityActivitiesFactoryItem,
+    [ReportingPage.AdditionalReportingData]: additionalReportingDataFactoryItem,
+    [ReportingPage.NewEntrantInformation]: newEntrantInformationFactoryItem,
+    [ReportingPage.OperationEmissionSummary]:
+      operationEmissionSummaryFactoryItem,
+    [ReportingPage.ComplianceSummary]: complianceSummaryFactoryItem,
+  };
+
+  let reviewData: ReviewData[] = [];
+  // Iterate over the provided flow's header steps and their associated pages
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for (const [headerStep, pages] of Object.entries(flow)) {
+    for (const page of pages) {
+      const factoryFn = factoryMapping[page];
+      if (factoryFn) {
+        let pageData: ReviewData[] = [];
+        // Determine parameters based on the ReportingPage
+        switch (page) {
+          case ReportingPage.Activities:
+            pageData = await factoryFn(versionId, currentFacilities, flow);
+            break;
+          default:
+            pageData = await factoryFn(versionId, facilityId);
+            break;
+        }
```

Comment:
we can pass the flow as a third param anyways - and make it part of the factory function type, the other functions wouldn't parse it anyways. 

In `factory.ts` the `ReviewDataFactoryItem` type gets that additional `flow` parameter, and only the activities factory item needs to be updated to use it

```suggestion
        pageData = await factoryFn(versionId, currentFacilities, flow);
```